### PR TITLE
Split common

### PR DESCRIPTION
**Pull request by That-Canadian**
@drashna @jackhumbert Fixed that I2C only bug (whoops) also added @nooges serial backlight implementation

### CODE REVIEW

**Comment by That-Canadian** [comment]
Ew I apologize for the commit log, no idea why it's showing up like that

File: quantum/split_common/matrix.c [hunk 1]
```diff
@@ -286,6 +286,16 @@ int serial_transaction(void) {
     for (int i = 0; i < ROWS_PER_HAND; ++i) {
         matrix[slaveOffset+i] = serial_slave_buffer[i];
     }
+    
+    #ifdef RGBLIGHT_ENABLE
+        // Code to send RGB over serial goes here (not implemented yet)
```
Comment (drashna) [inline quantum/split_common/matrix.c]: Well, for serial, probably using the second data line for the RGB strips, so probably isn't a big deal

File: quantum/split_common/split_util.c [hunk 2]
```diff
@@ -101,35 +101,46 @@ void keyboard_slave_loop(void) {
...
-            BACKLIT_DIRTY = false;
-        }
+        #ifdef USE_I2C
+            if (BACKLIT_DIRTY) {
+                backlight_set(i2c_slave_buffer[I2C_BACKLIT_START]);
+                BACKLIT_DIRTY = false;
+            }
+        #else // USE_SERIAL
+            if (BACKLIT_DIRTY) {
+                backlight_set(serial_master_buffer[SERIAL_BACKLIT_START]);
```
Comment (drashna) [inline quantum/split_common/split_util.c]: shouldn't `BACKLIT_DIRTY` be set to false after this, too?

Comment (That-Canadian) [reply quantum/split_common/split_util.c]: Yes, yes it should haha. This is why we have awesome reviewers like you ;)

File: quantum/split_common/matrix.c (hunk 1 above)
Comment (That-Canadian) [reply quantum/split_common/matrix.c]: True, but put the block there so if in the future it is implemented people know where to put it =P

Comment (That-Canadian) [reply quantum/split_common/matrix.c]: If you'd rather that removed also np, just let me know.

File: quantum/split_common/split_util.c (hunk 2 above)
Comment (drashna) [reply quantum/split_common/split_util.c]: Also, wouldn't this be cleaner: 
```c
    #ifdef BACKLIGHT_ENABLE
            if (BACKLIT_DIRTY) {
              #ifdef USE_I2C
                backlight_set(i2c_slave_buffer[I2C_BACKLIT_START]);
              #else // USE_SERIAL
                backlight_set(serial_master_buffer[SERIAL_BACKLIT_START]);
              #endif
              BACKLIT_DIRTY = false;
            }
    #endif
```

I mean, either works, but...

Comment (That-Canadian) [reply quantum/split_common/split_util.c]: Psh "cleaner"

but yes you're correct, that's a better way to do it, will update haha

Comment (drashna) [reply quantum/split_common/split_util.c]: Okay, "prettier" 😄

File: quantum/split_common/matrix.c (hunk 1 above)
Comment (drashna) [reply quantum/split_common/matrix.c]: No, it should be fine, I think. :)

**Comment by drashna** [comment]
For the commit log, it's because you're merging, and not rebasing, it looks like.

**Comment by drashna** [comment]
@jackhumbert This looks ready to be merged.

**Comment by jackhumbert** [comment]
Awesome - thanks folks!

**Pull request merged**